pkg/sql/catalog/descs/collection_test.go

```diff
@@ -284,7 +284,7 @@ func TestAddUncommittedDescriptorAndMutableResolution(t *testing.T) {
 			immByNameAfter, err := descriptors.GetImmutableDatabaseByName(ctx, txn, "new_name", flags)
 			require.NoError(t, err)
 			require.Equal(t, db.GetVersion(), immByNameAfter.GetVersion())
-			require.Equal(t, mut.ImmutableCopy(), immByNameAfter)
+			require.Equal(t, mut.ImmutableCopy().(catalog.DatabaseDescriptor).DatabaseDesc(), immByNameAfter.DatabaseDesc())
```

Comment:
nit: if you want to avoid the cast here, you can also compare the results of `.DescriptorProto()` to the same effect.